components/lib/n3-utils.js

```diff
@@ -0,0 +1,291 @@
+        });
+}
+
+/**
+ * Converts an N3 store into TTL, loads the TTL, and converts the TTL to JSON-LD.  This function
+ * is intended primarily for testing/double checking the n3ToJsonld method and will likely be 
+ * deleted or deprecated in the future. 
+ *
+ * @param store    (required) the N3 store to convert to text/Turtle format
+ * @param frame    (optional) a JSON-LD frame
+ * @param filename (optional) an optional file name to write the Turtle result into.
+ *
+ * @return a promise to return the N3 store converted to JSON-LD format or a reject on error
+ */ 
+function n3ToTtlToJsonld(store, frame, filename) { 
+
+    return new Promise(function(resolve, reject) { 
+
+        n3ToTtl(store).then(
+
+            function(ttl) { 
+                if (_.isEmpty(ttl)) { 
+                    return reject(errUtils.errorMessage("n3ToTtlToJsonld API received no content to convert to JSON-LD!"));
+                } 
+
+                ttl2Jsonld.ttlLoad(ttl).then(
+                    function(n3Store) { 
+
+                        var filterBnodeAttrs = [ 'id', '_id', '@id' ];
+                        ttl2Jsonld.rdfToJsonLd(n3Store, frame, filterBnodeAttrs).then(
+                            function(json) { 
+
+                                if (_.isEmpty(filename)) { 
+                                    return resolve(json);
+                                }
+
+                                fs.writeFile(filename, JSON.stringify(json, null, 2), 
+                                    function(error) { 
+                                        if (!_.isNull(error)) { 
+                                            return reject(errUtils.errorMessage("n3ToTtlToJsonld API was unable to write file " + filename + "!\n", error));
+                                        } 
+
+                                        return resolve(json);
+                                });
+                            });
+                         }, 
+
+                         function(fail) { 
+                             return reject(errUtils.errorMessage('Failed to load ttl!', fail));
+
+                     }).catch(function(e) { 
+                         return reject(errUtils.errorMessage('Exception loading ttl!', e));
+                     });
+            }, 
+
+            function(fail) { 
+                reject(errUtils.errorMessage("n3ToTtlToJsonld error converting N3 store to TTL!", fail));
+    
+        }).catch(function(e) { 
+            return Promise.reject(errUtils.errorMessage('Exception in n3ToTtlToJsonld!', e));
+        });
+    });
+}
+
+/**
+ * Converts an N3 store into JSON-LD. It is an asychronous function.
+ * 
+ * @param store    (required) the N3 store to convert to JSON-LD format
+ * @param frame    (required) a JSON-LD frame
+ * @param filename (optional) an optional file name to write the Turtle result into.
+ *
+ * @return a promise to return the N3 store converted to JSON-LD format or a reject on error
```

Comment:
Suggest rewording this to: @return a promise to return either the resulting JSON-LD or a reject on error